### PR TITLE
scripts: Typo in the word 'platform'

### LINKED ISSUE #58322: TestInstance create_overlay() extra_configs with per-platform filtering broken
**Describe the bug**
While expanding the test coverage for twister, I have noticed that platform-based filtering for extra configs in `testinstance.py`'s `create_overlay()` behaves in an unexpected manner. Instead of recognizing the colon-separated `platform` prefix as an indication to check for the platform, the whole config string is treated as a config.

**Expected behavior**
`TestInstance.create_overlay()` platform filtering for extra_configs to properly filter them.

**Impact**
TestInstance platform filtering for extra configs is broken completely.

**Logs and console output**
```
============================================================== FAILURES ===============================================================
___________________ test_create_overlay[False-False-False-coverage_platform11-native-extra_configs11-CONFIG_NULL=y] ___________________

class_testplan = <twisterlib.testplan.TestPlan object at 0x7fc364d51f00>
all_testsuites_dict = {'scripts/tests/twister/test_data/testsuites/samples/test_app/sample_test.app': <twisterlib.testsuite.TestSuite object...ster/test_data/testsuites/tests/test_b/test_b.check_1': <twisterlib.testsuite.TestSuite object at 0x7fc364e292a0>, ...}
platforms_list = [<demo_board_2 on x86_demo>, <demo_board_3 on arm_demo>, <demo_board_1 on arm_demo>], enable_asan = False
enable_ubsan = False, enable_coverage = False, coverage_platform = ['demo_board_2'], platform_type = 'native'
extra_configs = ['platform:demo_board_2:CONFIG_NULL=y'], expected_content = 'CONFIG_NULL=y'

    @pytest.mark.parametrize("enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, extra_configs, expected_content", TESTDATA_2)
    def test_create_overlay(class_testplan, all_testsuites_dict, platforms_list, enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, extra_configs, expected_content):
        """Test correct content is written to testcase_extra.conf based on if conditions."""
        class_testplan.testsuites = all_testsuites_dict
        testcase = class_testplan.testsuites.get('scripts/tests/twister/test_data/testsuites/samples/test_app/sample_test.app')

        if extra_configs:
            testcase.extra_configs = extra_configs

        class_testplan.platforms = platforms_list
        platform = class_testplan.get_platform("demo_board_2")

        testinstance = TestInstance(testcase, platform, class_testplan.env.outdir)
        platform.type = platform_type
>       assert testinstance.create_overlay(platform, enable_asan, enable_ubsan, enable_coverage, coverage_platform) == expected_content
E       AssertionError: assert 'platform:dem...CONFIG_NULL=y' == 'CONFIG_NULL=y'
E         - CONFIG_NULL=y
E         + platform:demo_board_2:CONFIG_NULL=y

tests/twister/test_testinstance.py:93: AssertionError
```

**Environment (please complete the following information):**
 - OS: Linux
 - N/A
 - Commit SHA: `28a18ece433ab1534392896acd7bb28bb3dd05a2`

### PR DESCRIPTION
**Pull request by LukaszMrugala**
`testinstance.py` contains a string literal comparison to the word `platform`, which has been misspelled as `plaform`.

That typo and a few others, contained within docstrings, have been corrected to `platform`.

The change is verified by a test expansion.

Fixes #58322

### CODE REVIEW

**Review by asemjonovs — DISMISSED** [review]

File: scripts/tests/twister/test_testinstance.py [hunk 1]
```diff
@@ -56,22 +56,30 @@ def test_check_build_or_run(class_testplan, monkeypatch, all_testsuites_dict, pl
...
-    """Test correct content is written to testcase_extra.conf based on if conditions
-    TO DO: Add extra_configs to the input list"""
+@pytest.mark.parametrize("enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, extra_configs, expected_content", TESTDATA_2)
+def test_create_overlay(class_testplan, all_testsuites_dict, platforms_list, enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, extra_configs, expected_content):
+    """Test correct content is written to testcase_extra.conf based on if conditions."""
     class_testplan.testsuites = all_testsuites_dict
     testcase = class_testplan.testsuites.get('scripts/tests/twister/test_data/testsuites/samples/test_app/sample_test.app')
+
+    if extra_configs:
+        testcase.extra_configs = extra_configs
```
Comment (golowanow) [inline scripts/tests/twister/test_testinstance.py:81]: @LukaszMrugala please split the `extra_configs` fix and the typo fix into separate commits and update the PR's description that there are two fixes here.

**Review by golowanow — CHANGES_REQUESTED** [review]

**Review by golowanow — DISMISSED** [review]

**Review by aasthagr — DISMISSED** [review]

**Comment by LukaszMrugala** [comment]
`CONFIG_NULL` changed to `CONFIG_LOG`. `CONFIG_NULL` is not defined anywhere, so it failed compliance checks. `CONFIG_LOG` has been deemed as close to a `null` config (one that does nothing), so it is used instead.

**Comment by nashif** [comment]
Can you please split the typo fixes in various tests into their own commits?

**Comment by LukaszMrugala** [comment]
> Can you please split the typo fixes in various tests into their own commits?

Yes, I could do that. I am assuming that you'd like one commit for `test_*.py` and `testinstance.py` change and then one commit for every `*.c` typo changed?

**Comment by nashif** [comment]
> Yes, I could do that. I am assuming that you'd like one commit for `test_*.py` and `testinstance.py` change and then one commit for every `*.c` typo changed?

no, one commit for the twister pytest tests, another for tests/ztest/.. and one more for test/lib/...

**Review by nashif — APPROVED** [review]

**Comment by golowanow** [comment]
@LukaszMrugala the two new commits have identical description, could change subjects there like this:
`tests: lib: devicetree: Fixes docstring typo`
`tests: shared_multi_heap: Fixes docstring typo`

**Review by hakehuang — APPROVED** [review]

**Review by golowanow — APPROVED** [review]

**Pull request merged**